lib/hanami/cli/registry.rb

```diff
@@ -0,0 +1,46 @@
+require "hanami/cli/command_registry"
+
+module Hanami
+  class Cli
+    module Registry
+      def self.extended(base)
+        base.class_eval do
+          @commands = CommandRegistry.new
+        end
+      end
+
+      def register(name, command = nil, aliases: [])
+        if block_given?
+          yield Prefix.new(@commands, name, aliases)
+        else
+          command.command_name = name if Cli.command?(command)
```

Comment:
Perhaps a better way to explain why I think it might be useful to allow instantiation of commands is if you have the same command class but with a little bit of configuration, it can perform 2 different things in your CLI. So in this case you'd want to register it twice with these different options:

```ruby
register "thing",         MyApp::CLI::ThingCommand
register "another_thing", MyApp::CLI::ThingCommand, be_slightly_different: true
```

If we stuck to only registering whole-classes, then the only way to provide this "same behaviour but supporting different config" would be to have the command author split it across two separate classes, which feels like a bit of a hack.